### PR TITLE
 Pull #5113: enabled eclipse compiler to flag unused exceptions 

### PR DESCRIPTION
**Pull request by rnveach**
Testing if eclipse can be configured to flag unused exceptions in tests.

### CODE REVIEW

**Comment by rnveach** [comment]
It doesn't seem to flag anything from the test directories as I am getting violations in my eclipse even on other things besides unused exceptions.
I will need to look into the reason. I am thinking the classpath isn't receiving the tests.

**Comment by rnveach** [comment]
Along with the classpath being changed, [the shell script has to be modified to include the test folder](https://github.com/checkstyle/checkstyle/blob/27416796f6143a4ca8d9fc008685d0c13e25c79a/.ci/eclipse-compiler-javac.sh#L28). 

When doing so, it results in ~24 errors in master which I am also seeing locally inside Eclipse.
Most are `The allocated object is never used`, `The declared exception CheckstyleException is not actually thrown`, constructor/methods never being used, etc...

**Comment by rnveach** [comment]
This PR is dependent on https://github.com/checkstyle/checkstyle/pull/5143 first.

File: config/org.eclipse.jdt.core.prefs [hunk 1]
```diff
@@ -44,6 +44,9 @@ org.eclipse.jdt.core.compiler.problem.unlikelyCollectionMethodArgumentType=error
 org.eclipse.jdt.core.compiler.problem.unlikelyCollectionMethodArgumentTypeStrict=enabled
 org.eclipse.jdt.core.compiler.problem.unlikelyEqualsArgumentType=error
 org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownException=error
+org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionExemptExceptionAndThrowable=disabled
```
Comment (rnveach) [inline config/org.eclipse.jdt.core.prefs]: This option in Eclipse is "Ignore Exception and Throwable".

File: config/org.eclipse.jdt.core.prefs [hunk 2]
```diff
@@ -44,6 +44,9 @@ org.eclipse.jdt.core.compiler.problem.unlikelyCollectionMethodArgumentType=error
...
+org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionIncludeDocCommentReference=enabled
```
Comment (rnveach) [inline config/org.eclipse.jdt.core.prefs]: This option in Eclipse is "Ignore exceptions documented with `'@throws'` or `'@exception'` tags".

File: config/org.eclipse.jdt.core.prefs [hunk 3]
```diff
@@ -44,6 +44,9 @@ org.eclipse.jdt.core.compiler.problem.unlikelyCollectionMethodArgumentType=error
...
+org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionWhenOverriding=disabled
```
Comment (rnveach) [inline config/org.eclipse.jdt.core.prefs]: This option in Eclipse is "Ignore in overriding and implementing methods".

**Comment by rnveach** [comment]
@romani This PR is ready for review.

**Comment by rnveach** [comment]
@romani done.

**Comment by romani** [comment]
Travis failure is valid:
```
1. ERROR in /home/travis/build/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/utils/FilterUtilsTest.java (at line 61)
	public void testNonExistingFile() throws Exception {
	                                         ^^^^^^^^^
The declared exception Exception is not actually thrown by the method testNonExistingFile() from type FilterUtilsTest
```

**Comment by rnveach** [comment]
@romani Fixed.

**Comment by romani** [comment]
Travis failures were unrelated

**Pull request merged**